src/analysis/convex/basic.lean

```diff
@@ -190,6 +193,30 @@ hs.linear_preimage $ hf.mk' f
 lemma convex.add {t : set E} (hs : convex 𝕜 s) (ht : convex 𝕜 t) : convex 𝕜 (s + t) :=
 by { rw ← add_image_prod, exact (hs.prod ht).is_linear_image is_linear_map.is_linear_map_add }
 
+variables (𝕜 E)
+
+/-- The convex sets form an additive submonoid under pointwise addition. -/
+def convex_add_submonoid : add_submonoid (set E) :=
+{ carrier := {s : set E | convex 𝕜 s},
+  zero_mem' := convex_zero,
+  add_mem' := λ s t, convex.add }
+
+variables {𝕜 E}
+
+@[simp] lemma mem_convex_add_submonoid {s : set E} :
+  s ∈ convex_add_submonoid 𝕜 E ↔ convex 𝕜 s :=
+iff.rfl
+
+lemma convex_list_sum (s : list (set E)) (h : ∀ i ∈ s, convex 𝕜 i) : convex 𝕜 s.sum :=
```

Comment:
```suggestion
lemma convex_list_sum {s : list (set E)} (h : ∀ i ∈ s, convex 𝕜 i) : convex 𝕜 s.sum :=
```
and below. What do you think?

Reply:
I don't like it, but I guess it's consistent with the `add_submonoid` API.